src/kwlib.sh

```diff
@@ -265,41 +265,49 @@ function join_path()
 # This function checks if the target distro is supported by kw. This function
 # is handy for plugins that have some dependency with distros.
 #
-# Note: We handle OS family instead of a specific distro_id, for example,
-# Ubuntu and Mint are Debian based distro_id.
+# Accordingly, with os-release documentation, when we find ID_LIKE, we are in a
+# derivative distro. If we don't have it, the distro is the original one and
+# will only have ID.
 #
-# @root_path Expects the root path wherein we can find the /etc
-# @str_check String with a distro name
+# @root_path: Expects the root path wherein we can find the /etc
+# @str_check: String with a distro name
+# @raw_os_release: os-release file in a string
 #
 # Returns:
 # It returns the family name in lowercase, otherwise return none.
 function detect_distro()
 {
   local root_path="$1"
   local str_check="$2"
+  local raw_os_release="$3"
   local distro_id='none'
   local etc_path
-  declare -a debian_family=('debian' 'ubuntu' 'raspbian')
-  declare -a arch_family=('arch' 'manjaro')
-  declare -a fedora_family=('fedora')
+  local os_release_processed
+  declare -a os_family=('debian' 'arch' 'fedora')
 
-  etc_path=$(join_path "$root_path" /etc)
+  etc_path=$(join_path "$root_path" '/etc')
+
+  if [[ -d "$etc_path" && -z "$str_check" && -z "$raw_os_release" ]]; then
+    os_release_process=$(< "${etc_path}/os-release")
+  elif [[ -n "$raw_os_release" ]]; then
+    os_release_process="$raw_os_release"
+  fi
+
+  if [[ -n "$os_release_process" ]]; then
+    distro_id=$(printf '%s' "$os_release_process" | grep -w ID_LIKE | cut -d = -f 2)
+
+    # It is not a derivative distro; let's check ID.
+    if [[ -z "$distro_id" ]]; then
+      distro_id=$(printf '%s' "$os_release_process" | grep -w ID | cut -d = -f 2)
+    fi
+  fi
 
   if [[ -n "$str_check" ]]; then
     distro_id="$str_check"
-  elif [[ -d $etc_path ]]; then
-    distro_id=$(cat "$etc_path"/*-release | grep -w ID | cut -d = -f 2)
   fi
 
-  # Debian family
-  if [[ "${debian_family[*]}" =~ ${distro_id} ]]; then
-    printf '%s\n' 'debian'
-  # ArchLinux family
-  elif [[ "${arch_family[*]}" =~ ${distro_id} ]]; then
-    printf '%s\n' 'arch'
-  # Fedora family
-  elif [[ "${fedora_family[*]}" =~ ${distro_id} ]]; then
-    printf '%s\n' 'fedora'
+  if [[ ${os_family[*]} =~ ${distro_id} ]]; then
```

Comment:
It may be worth taking into consideration that `ID_LIKE` might have more than one operating system listed. According to the manpage:

```
ID_LIKE=
A space-separated list of operating system identifiers in the same syntax as the ID= setting.
```
For instance, Pop!_OS has `ID_LIKE="ubuntu debian"`, and then this comparison is evaluated as false.

Reply:
Based on this comment I think that a for loop can be constructed by creating a `$distro_ids` like so:
```bash
distro_ids=$(printf '%s' "$os_release_processed" | grep -w 'ID\(_LIKE\)\?' | cut -d = -f 2)
```
This gathers all entries, from both `ID` and `ID_LIKE`, then it's possible to loop over the entries with:
```bash
for distro_id in $distro_ids; do
  if [[ ${os_family[*]} =~ ${distro_id} ]]; then
    printf '%s\n' "$distro_id"
    return
  fi
done

printf '%s\n' 'none'
```